lib/streamlit/runtime/secrets.py

```diff
@@ -51,20 +47,90 @@ def _convert_to_dict(obj: Mapping[str, Any] | AttrDict) -> dict[str, Any]:
     return {k: v.to_dict() if isinstance(v, AttrDict) else v for k, v in obj.items()}
 
 
+"""
+SecretErrorMessages stores all error messages we use for secrets to allow customization for different environments.
+For example Streamlit Cloud can customize the message to be different than the open source one.
+"""
+
+
+class SecretErrorMessagesKey(Enum):
+    MISSING_ATTR_MESSAGE = "MISSING_ATTR_MESSAGE"
+    MISSING_KEY_MESSAGE = "MISSING_KEY_MESSAGE"
+    NO_SECRETS_FOUND = "NO_SECRETS_FOUND"
+    ERROR_PARSING_FILE_AT_PATH = "ERROR_PARSING_FILE_AT_PATH"
+    SUBFOLDER_PATH_IS_NOT_A_FOLDER = "SUBFOLDER_PATH_IS_NOT_A_FOLDER"
+    INVALID_SECRET_PATH = "INVALID_SECRET_PATH"
+
+
+class SecretErrorMessages:
+    def __init__(self):
+        self.messages = {}
+        self.set_message(
+            SecretErrorMessagesKey.MISSING_ATTR_MESSAGE,
+            lambda attr_name: (
+                f'st.secrets has no attribute "{attr_name}". '
+                f"Did you forget to add it to secrets.toml, mount it to secret directory, or the app settings on Streamlit Cloud? "
+                f"More info: https://docs.streamlit.io/deploy/streamlit-community-cloud/deploy-your-app/secrets-management"
+            ),
+        )
```

Comment:
great idea

Reply:
Was going to agree with @vdonato. Is there a reason we don't just do one of the following:
1. Just define functions for each of these messages
2. Consider maybe a class with methods?

I'm struggling to see how ` secret_error_messages_singleton.get_message(SecretErrorMessagesKey.ERROR_PARSING_FILE_AT_PATH)(path)`

can't just be `get_error_parsing_file_at_path(path)` or `messages.get_error_parsing_file_at_path(path)` ?

Also the types in the lambda arguments can be different, so this would also ensure we can enforce these correctly?

Reply:
i can see the benefits. changed so it uses getter / setter style.